### PR TITLE
cmd/atlas: make schema diff commands work with realms

### PR DESCRIPTION
**Pull request by ronenlu**
currently the command: `atlas schema diff --from mysql://root:pass@localhost:3306 --to mysql://root:pass@localhost:33`
will return: `Error: mysql: schema "" was not found`

### CODE REVIEW

File: cmd/atlas/internal/cmdapi/diff.go [hunk 1]
```diff
@@ -54,19 +55,30 @@ func cmdDiffRun(cmd *cobra.Command, flags *diffCmdOpts) {
...
-	toSchema, err := toC.InspectSchema(ctx, toName, nil)
-	cobra.CheckErr(err)
-	// SchemaDiff checks for name equality which is irrelevant in the case
-	// the user wants to compare their contents, if the names are different
-	// we reset them to allow the comparison.
-	if fromName != toName {
-		toSchema.Name = ""
-		fromSchema.Name = ""
+	var diff []schema.Change
+	// if one of schema names is empty we compare realms
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/diff.go:59]: Add validation that both connections/urls are the same type. Schema or a realm.

File: cmd/atlas/internal/cmdapi/diff.go [hunk 2]
```diff
@@ -54,19 +56,33 @@ func cmdDiffRun(cmd *cobra.Command, flags *diffCmdOpts) {
...
+	if fromName == "" || toName == "" {
+		if fromName != toName {
+			cobra.CheckErr(fmt.Errorf("can not diff realm with schame, from schema: %s, to schema: %s", fromName, toName))
```
Comment (masseelch) [inline cmd/atlas/internal/cmdapi/diff.go:63]:
```suggestion
			cobra.CheckErr(fmt.Errorf("can not diff realm with schema, from schema: %s, to schema: %s", fromName, toName))
```

File: cmd/atlas/internal/cmdapi/diff.go [hunk 3]
```diff
@@ -54,19 +56,33 @@ func cmdDiffRun(cmd *cobra.Command, flags *diffCmdOpts) {
 	defer toC.Close()
 	fromName := fromC.URL.Schema
 	toName := toC.URL.Schema
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/diff.go:58]: toName and fromName and not good names anymore. Use `fromS`, `toS` instead.

File: cmd/atlas/internal/cmdapi/diff.go [hunk 4]
```diff
@@ -54,19 +56,33 @@ func cmdDiffRun(cmd *cobra.Command, flags *diffCmdOpts) {
...
-	cobra.CheckErr(err)
-	// SchemaDiff checks for name equality which is irrelevant in the case
-	// the user wants to compare their contents, if the names are different
-	// we reset them to allow the comparison.
-	if fromName != toName {
-		toSchema.Name = ""
-		fromSchema.Name = ""
+	var diff []schema.Change
+	// if one of schema names is empty we compare realms
+	if fromName == "" || toName == "" {
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/diff.go:61]: switch statements are powerful in Go, use it here to make the code simpler.

```go
switch {
case fromS == "" && toS == "":
// compare realm.
case fromS == "":
	return fmt.Errorf("cannot diff schema %q with a database connection", fromS)
case toS == "":
	return fmt.Errorf("cannot diff a database connection with a schema %q ", toS)
default: // compare schemas.
}
```

**Review by a8m — APPROVED** [review]
💯 

Can you add testscript for this in another PR? Let me know if you need any help.

**Pull request merged**